tests/e2e/config/bootstrap.js

```diff
@@ -0,0 +1,185 @@
+ */
+function removePageEvents() {
+  pageEvents.forEach(([eventName, handler]) => {
+    page.removeListener(eventName, handler);
+  });
+}
+
+/**
+ * Adds a page event handler to emit uncaught exception to process if one of
+ * the observed console logging types is encountered.
+ */
+function observeConsoleLogging() {
+  page.on('console', (message) => {
+    const type = message.type();
+    if (!OBSERVED_CONSOLE_MESSAGE_TYPES.hasOwnProperty(type)) {
+      return;
+    }
+
+    let text = message.text();
+
+    // styled-components warns about dynamically created components.
+    // @todo Fix issues.
+    if (text.includes(' has been created dynamically.')) {
+      return;
+    }
+
+    const logFunction = OBSERVED_CONSOLE_MESSAGE_TYPES[type];
+
+    // As of Puppeteer 1.6.1, `message.text()` wrongly returns an object of
+    // type JSHandle for error logging, instead of the expected string.
+    //
+    // See: https://github.com/GoogleChrome/puppeteer/issues/3397
+    //
+    // The recommendation there to asynchronously resolve the error value
+    // upon a console event may be prone to a race condition with the test
+    // completion, leaving a possibility of an error not being surfaced
+    // correctly. Instead, the logic here synchronously inspects the
+    // internal object shape of the JSHandle to find the error text. If it
+    // cannot be found, the default text value is used instead.
+    text = message.args()?.[0]?._remoteObject?.description || text;
+
+    // Disable reason: We intentionally bubble up the console message
+    // which, unless the test explicitly anticipates the logging via
+    // @wordpress/jest-console matchers, will cause the intended test
+    // failure.
+
+    // eslint-disable-next-line no-console
+    console[logFunction](text);
+  });
+}
+
+/**
+ * Runs Axe tests when the story editor is found on the current page.
+ *
+ * @return {?Promise} Promise resolving once Axe texts are finished.
+ */
+async function runAxeTestsForStoriesEditor() {
+  if (!(await page.$('body.edit-story'))) {
+    return;
+  }
+
+  await expect(page).toPassAxeTests({
+    // Temporary disabled rules to enable initial integration.
+    disabledRules: [
+      'aria-allowed-role',
+      'aria-input-field-name',
+      'aria-required-parent',
+      'button-name',
+      'color-contrast',
+      'label',
+      'landmark-banner-is-top-level',
+      'landmark-no-duplicate-banner',
+      'landmark-unique',
+      'page-has-heading-one',
+      'region',
+    ],
+  });
+}
+
+/**
+ * Before every test suite run, delete all content created by the test. This ensures
+ * other posts/comments/etc. aren't dirtying tests and tests don't depend on
+ * each other's side-effects.
+ */
+// eslint-disable-next-line jest/require-top-level-describe
+beforeAll(async () => {
+  capturePageEventsForTearDown();
+  enablePageDialogAccept();
+  observeConsoleLogging();
+  // 15inch screen.
+  await setBrowserViewport({
```

Comment:
This code is repeated. Why not implement something like [setupBrowser](https://github.com/WordPress/gutenberg/blob/master/packages/e2e-tests/config/setup-test-framework.js#L64-L67).

Reply:
Done in 8dfda8f82e95a133b2bf96d26ed721dcebe37580.